demos/ensembles/ensemble-link-prediction-example.ipynb

```diff
@@ -321,6 +321,7 @@
       "\n",
```

Comment:
<p>I'm missing a bit of context for why this had <code>val_weighted_acc</code>  in the first place, when it apparently doesn't exist. Possibly worth checking with @PantelisElinas?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1008/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

Reply:
yeah I'm not sure but it seems like a typo and I think this warning is new so maybe it was never picked up. Either way I think using the `val_acc` is reasonable here

Reply:
okay sure 👍

Reply:
Hmm I don't see these early stopping parameters in the tensorflow docs for `Model` at all actually 🤔 https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit_generator
or even in the 1.15 docs https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/keras/Model#fit_generator

Does it seem like it's actually doing anything currently...?

The early stopping docs for tensorflow looks very different: https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/EarlyStopping

Reply:
okay checked this out and just realised that `model` here is an instance of our custom `BaggingEnsemble` not `tf.keras.Model`, and `BaggingEnsemble.fit_generator` supports this argument

Reply:
Oops thanks for figuring that out! 

I do feel like this is potentially a bug though - in an earlier cell it looks like when we compile the model, we use
```python
model.compile(
    optimizer=keras.optimizers.Adam(lr=1e-3),
    loss=keras.losses.binary_crossentropy,
    weighted_metrics=["acc"],
)
```
so `val_weighted_acc` should be available, but it's not? And if we use `val_acc` instead, are we supposed to be passing in `metrics=["acc"]` when we compile the model?

The existing warning message said:
```
WARNING:tensorflow:Early stopping conditioned on metric `val_weighted_acc` which is not available. Available metrics are: loss,acc,val_loss,val_acc
```

so it looks to me like something strange seems to be going on, where the `weighted_metrics` parameter ends up creating a normal metric maybe..? I'm happy for it to be a separate issue though, since the warnings have been removed.

Reply:
Seems like `tf` is still using weighted accuracy but just isn't adding `"weighted_"` to the string.

Reply:
Yeah looks like it's probably a "feature"
```
>>> import tensorflow as tf
>>> x_in = tf.keras.layers.Input(1)
>>> x_out = tf.keras.layers.Dense(1)(x_in)
>>> model = tf.keras.Model(inputs=x_in, outputs=x_out)
>>> model.compile(
...     optimizer='adam',
...     loss='categorical_crossentropy',
...     weighted_metrics=["acc"],
...     metrics=["acc"]
... )
>>> print(model.metrics_names)
['loss', 'acc', 'acc_1']
```